src/map/script.cpp

```diff
@@ -6774,7 +6774,7 @@ BUILDIN_FUNC(countinarray)
 
 	i = reference_getindex(data1);
 	j = reference_getindex(data2);
-	if (array_size1 < i || array_size2 < j)
+	if (array_size1 - 1 < i || array_size2 - 1 < j)
```

Comment:
Can we change this check to `if (i + 1 > array_size1 || j + 1 > array_size2)`?
This is checking if the reference that we passed in is outside the array 
> for example, `.@arr[10]`
if the highest key is 2, then `.@arr[10]` is outside the size.

Reply:
Couldn't this overflow?

Reply:
You're right, it could overflow.